File: app/src/main/java/com/example/android/sunshine/utilities/DataBaseChecker.java
package com.example.android.sunshine.utilities;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.Observer;

import com.example.android.sunshine.AppExecutors;
import com.example.android.sunshine.data.SunshineRepository;
import com.example.android.sunshine.data.database.ListViewWeatherEntry;
import com.example.android.sunshine.data.database.WeatherEntry;

import java.util.List;

public class DataBaseChecker extends AppCompatActivity {


    private SunshineRepository mRepository;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        mRepository = InjectorUtils.provideRepository(this);
        showDataBase();


    }


    private void showDataBase() {
        AppExecutors.getInstance().diskIO().execute(new Runnable() {
            @Override
            public void run() {
                for (WeatherEntry weatherEntry : mRepository.mWeatherDao.getAll()) {
                    L.f("Date : " + weatherEntry.getDate());
                }


                LiveData<List<ListViewWeatherEntry>> currentWeatherForecasts = mRepository.mWeatherDao.getCurrentWeatherForecasts(SunshineDateUtils.getNormalizedUtcDateForToday());

                AppExecutors.getInstance().mainThread().execute(new Runnable() {
                    @Override
                    public void run() {
                        currentWeatherForecasts.observe(DataBaseChecker.this, new Observer<List<ListViewWeatherEntry>>() {
                            @Override
                            public void onChanged(List<ListViewWeatherEntry> weatherEntries) {
                                if (weatherEntries == null)
                                    return;

                                for (ListViewWeatherEntry weatherEntry : weatherEntries) {
                                    L.f("Future Date : " + weatherEntry.getDate());

                                }
                            }
                        });
                    }
                });


                //List<WeatherEntry> value = mRepository.mWeatherDao.getCurrentWeatherForecastsTest(SunshineDateUtils.getNormalizedUtcDateForToday());
                List<ListViewWeatherEntry> value = mRepository.mWeatherDao.getCurrentWeatherForecasts(SunshineDateUtils.getNormalizedUtcDateForToday()).getValue();

                if (value == null)
                    return;

                for (ListViewWeatherEntry weatherEntry : value) {
                    L.f("Future Date : " + weatherEntry.getDate());

                }

            }
        });


    }
}

File: app/src/main/java/com/example/android/sunshine/utilities/DataBaseChecker.kt
package com.example.android.sunshine.utilities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer

import com.example.android.sunshine.AppExecutors
import com.example.android.sunshine.data.SunshineRepository

class DataBaseChecker : AppCompatActivity() {


    private var mRepository: SunshineRepository? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)


        mRepository = InjectorUtils.provideRepository(this)
        showDataBase()


    }


    private fun showDataBase() {
        AppExecutors.instance.diskIO().execute(Runnable {
            for (weatherEntry in mRepository!!.mWeatherDao.all) {
                L.f("Date : " + weatherEntry.date!!)
            }


            val currentWeatherForecasts = mRepository!!.mWeatherDao.getCurrentWeatherForecasts(SunshineDateUtils.normalizedUtcDateForToday)

            AppExecutors.instance.mainThread().execute(Runnable {
                currentWeatherForecasts.observe(this@DataBaseChecker, Observer { weatherEntries ->
                    if (weatherEntries == null)
                        return@Observer

                    for (weatherEntry in weatherEntries) {
                        L.f("Future Date : " + weatherEntry.date)

                    }
                })
            })


            //List<WeatherEntry> value = mRepository.mWeatherDao.getCurrentWeatherForecastsTest(SunshineDateUtils.getNormalizedUtcDateForToday());
            val value = mRepository!!.mWeatherDao.getCurrentWeatherForecasts(SunshineDateUtils.normalizedUtcDateForToday).value
                    ?: return@Runnable

            for (weatherEntry in value) {
                L.f("Future Date : " + weatherEntry.date)

            }
        })


    }
}
